utilities/xmlrpcpp/src/XmlRpcValue.cpp

```diff
@@ -610,16 +611,52 @@ namespace XmlRpc {
       default:           break;
       case TypeBoolean:  os << _value.asBool; break;
       case TypeInt:      os << _value.asInt; break;
-      case TypeDouble:   os << _value.asDouble; break;
+      case TypeDouble:
+        {
+          static std::once_flag once;
+          int ret, required_size;
+          char buf[128]; // Should be long enough
+          ret = std::snprintf(buf, sizeof(buf)-1,
+                  getDoubleFormat().c_str(), _value.asDouble);
+          if (ret < 0) {
+            std::call_once(once,
+              [](){XmlRpcUtil::error("Failed to format with %s, using default %%.16g format",
+                getDoubleFormat().c_str());});
+            required_size = std::snprintf(buf, sizeof(buf)-1, "%.16g", _value.asDouble);
+          } else {
+            required_size = ret;
+          }
+          if (required_size >= 0 && required_size < (int) sizeof(buf)) {
+            buf[sizeof(buf)-1] = 0;
+            os << buf;
+          } else if (required_size >= (int) sizeof(buf)) {
+            int ret;
+            char required_buf[required_size+1];
+            ret = std::snprintf(required_buf, required_size,
+                    getDoubleFormat().c_str(), _value.asDouble);
+            if (ret == required_size) {
+              required_buf[required_size] = 0;
+              os << required_buf;
+            }
+          } else {
+            XmlRpcUtil::error("Unexpected error to format %s", getDoubleFormat().c_str());
```

Comment:
This message would point to the user provided format string - even when the above logic has switched to use the default format string.

Reply:
fixed.